lib/api-request.js

```diff
@@ -285,8 +285,27 @@ APIRequest.prototype._retry = function(err) {
 	this.numRetries = this.numRetries || 0;
 
 	if (this.numRetries < this.config.numMaxRetries) {
+		var retryTimeout;
 		this.numRetries += 1;
-		setTimeout(this.execute.bind(this), getRetryTimeout(this.numRetries, this.config.retryIntervalMS));
+		if (this.config.retryStrategy) {
+			var retryOptions = {
+				error: err,
+				numRetries: this.numRetries,
+				numMaxRetries: this.config.numMaxRetries,
+				retryIntervalMS: this.config.retryIntervalMS
```

Comment:
We should definitely add the number of attempts so far here, and also the total number of ms spent waiting if possible